### PR TITLE
Cluster

### PR DESCRIPTION
**Pull request by marianogonzalez**

### CODE REVIEW

**Comment by estebanwasinger** [comment]
PR name

File: src/main/java/org/mule/runtime/api/meta/model/declaration/fluent/SourceDeclaration.java [hunk 1]
```diff
@@ -120,6 +121,25 @@ public void setTerminateCallback(SourceCallbackDeclaration terminateCallback) {
     this.terminateCallback = terminateCallback;
   }
 
+  public boolean isHasResponse() {
```
[the remark this answers is not in the document]
Comment (marianogonzalez) [reply src/main/java/org/mule/runtime/api/meta/model/declaration/fluent/SourceDeclaration.java]: too late. This was already released in 1.0

File: src/main/java/org/mule/runtime/api/meta/model/declaration/fluent/SourceDeclaration.java [hunk 2]
```diff
@@ -120,6 +121,25 @@ public void setTerminateCallback(SourceCallbackDeclaration terminateCallback) {
...
+    return hasResponse;
+  }
+
+  /**
+   * @return whether the declared source should only run in the primary node when in cluster mode
+   */
+  public boolean isRunsOnPrimaryNodeOnly() {
```
Comment (alegmarra) [inline src/main/java/org/mule/runtime/api/meta/model/declaration/fluent/SourceDeclaration.java]: runsOnPrimaryNodeOnly

File: src/main/java/org/mule/runtime/api/meta/model/source/SourceModel.java [hunk 3]
```diff
@@ -66,6 +69,11 @@
    */
   Optional<SourceCallbackModel> getTerminateCallback();
 
+  /**
+   * @return Whether the declared source should only run in the primary node when in cluster mode
+   */
+  boolean isRunsOnPrimaryNodeOnly();
```
Comment (alegmarra) [inline src/main/java/org/mule/runtime/api/meta/model/source/SourceModel.java]: runsOnPrimaryNodeOnly

**Pull request merged**